Source/TexturePool.h

```diff
@@ -112,14 +111,21 @@ class TexturePool
 
         if (desc.Width == desc_shared.Width && desc.Height == desc_shared.Height && desc.Format == desc_shared.Format)
         {
-            pool.push_back(texture);
-            texture->AddRef();
+            pool.push_back(WrapTexturePointer(texture));
         }
     }
 
 private:
-    ID3D11Device* device = NULL;
-    std::vector<ID3D11Texture2D*> pool;
+
+    winrt::com_ptr<ID3D11Texture2D> WrapTexturePointer(ID3D11Texture2D* texture)
```

Comment:
Are you sure we need this? I'd assume that we can just directly store the com_ptr in the vector and retrieve it.

Reply:
We still need to do this eventually. Ffmpeg avframes return pointers to Texture2D, so if we do this here or when we return the texture to the pool we need to do it anyway. This was a least impactful change.

Reply:
Oh only now I realize that we get a pointer in here, not a com_ptr. Ok. This could probably be simplified when we completely switch to com_ptr.

Reply:
I tried going down that route the first time around and I don't see how we can trick that, unless I am missing something.